p2p/p2p.go

```diff
@@ -0,0 +1,69 @@
+package p2p
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.nanomsg.org/mangos/v3/protocol/bus"
+	_ "go.nanomsg.org/mangos/v3/transport/tcp"
+	"go.uber.org/zap"
+
+	"github.com/Zamerykanizowana/replicated-file-system/config"
+)
+
+const transportScheme = "tcp"
+
+type peer struct {
+	Address string
+	Name    string
+}
+
+func Run(selfConfig *config.PeerConfig, peersConfig []*config.PeerConfig) error {
+	sock, err := bus.NewSocket()
+	if err != nil {
+		zap.L().With(zap.Error(err)).Fatal("failed to setup BUS protocol socket")
+	}
+	selfURL := buildURL(selfConfig)
+	if err = sock.Listen(selfURL.String()); err != nil {
+		zap.L().With(zap.Error(err)).Fatal("failed to start listening on the BUS")
+	}
+
+	peers := make([]peer, 0, len(peersConfig))
+	for _, p := range peersConfig {
+		peerURL := buildURL(p)
+		peers = append(peers, peer{
+			Address: peerURL.String(),
+			Name:    p.Name,
+		})
+	}
+
+	time.Sleep(time.Second)
```

Comment:
Are those sleeps purely for experimentation here or is there some merit to them? :smile:

Reply:
just experiments :)